dask/delayed.py

```diff
@@ -208,11 +209,15 @@ def delayed(obj, name=None, pure=False):
     """
     if isinstance(obj, Delayed):
         return obj
+
     task, dasks = to_task_dasks(obj)
-    root = not dasks
-    name = name or '%s-%s' % (type(obj).__name__, tokenize(task, pure=True))
-    dasks.append({name: task})
-    return Delayed(name, dasks, pure=pure, root=root)
+
+    if not dasks and callable(obj):
+        return DelayedFunction(obj, pure=pure, prefix=prefix)
```

Comment:
Sure. I guess it's just a difference between how we divide the two uses of `delayed`. I see it as wrapping either a thing with dask graphs, resulting in a `Delayed`, or some immediate python object, resulting in a `DelayedRoot` (the root indicating it's at the base of the dag, there's probably a better name).

The special-casing of `__call__` for immediate functions is really just one of the things you can special case for any immediate object. For example, one could also do attribute/method checking at graph build time. So I don't see a distinction between callable/non-callable, rather I see one between lazy/immediate. And I find it easier to describe this dichotomy rather than one between callable objects that don't contain dasks/everything else.

As an aside, for the keyword thing perhaps drop `prefix` and just use `name`/part of name instead? Or is there a good reason to have these be separate?